ibis/backends/clickhouse/__init__.py

```diff
@@ -184,26 +302,136 @@ def _client_execute(self, query, external_tables=None):
                 external_tables=external_tables,
             )
 
+    def to_pyarrow_batches(
+        self,
+        expr: ir.Expr,
+        *,
+        params: Mapping[ir.Scalar, Any] | None = None,
+        limit: int | str | None = None,
+        chunk_size: int = 1_000_000,
+        **kwargs: Any,
+    ) -> pa.RecordBatchReader:
+        """Execute expression and return results in an iterator of pyarrow
+        record batches.
+
+        This method is eager and will execute the associated expression
+        immediately.
+
+        Parameters
+        ----------
+        expr
+            Ibis expression to export to pyarrow
+        limit
+            An integer to effect a specific row limit. A value of `None` means
+            "no limit". The default is in `ibis/config.py`.
+        params
+            Mapping of scalar parameter expressions to value.
+        chunk_size
+            Number of rows in each returned record batch.
+
+        Returns
+        -------
+        results
+            RecordBatchReader
+        """
+        pa = self._import_pyarrow()
+
+        from ibis.backends.pyarrow.datatypes import ibis_to_pyarrow_struct
+
+        schema = self._table_or_column_schema(expr)
+
+        def _batches():
+            for batch in self._cursor_batches(
+                expr, params=params, limit=limit, chunk_size=chunk_size
+            ):
+                struct_array = pa.array(
+                    map(tuple, batch),
```

Comment:
Is this conversion required? What's the previous type?

Reply:
This is copypasted from the `BaseSQLBackend` implementation. We'll have to do another cleaning once our compilers are all based on `sqlglot`.

Reply:
Please add a TODO note or a follow-up issue so we don't forget it.